benchmarks/python/core.py

```diff
@@ -228,6 +244,14 @@ def torchprofile_timer(self) -> float:
             self.prof.start()
         return self.current_time
 
+    def fusionprofile_timer(self) -> float:
+        if not self.execution_start:
+            profile = self.fd.profile()
+            elapsed_host_time = profile.host_time_ms / 1e3
+            self._increment_global_time(elapsed_host_time)
+        self.execution_start = not self.execution_start
```

Comment:
Can you explain in a comment why we need to toggle `self.execution_start` here?

Reply:
`fd.profile()` cannot be queried before `fd.execute`.
Pytest has the following structure when running the benchmark:
```
timer() # Cannot query fd.profile() here. It will raise an error
output = benchmark_fn()
timer() # Query fd.profile() here and increment clock.
```
The alternative is to use a `try-except` block.